library/alloc/src/str.rs

```diff
@@ -368,21 +368,17 @@ impl str {
     pub fn to_lowercase(&self) -> String {
         let out = convert_while_ascii(self.as_bytes(), u8::to_ascii_lowercase);
 
-        // Safety: we know this is a valid char boundary since
-        // out.len() is only progressed if ascii bytes are found
-        let rest = unsafe { self.get_unchecked(out.len()..) };
-
         // Safety: We have written only valid ASCII to our vec
         let mut s = unsafe { String::from_utf8_unchecked(out) };
 
-        for (i, c) in rest[..].char_indices() {
+        for (i, c) in self.char_indices().skip(s.len()) {
```

Comment:
Good point, fixed thanks!
In the new version, since `out` is moved, I recompute `out.len()` from `self.len() - rest.len()`. The 'Σ' case is unlikely so I chose not to copy `out.len()`. But if you think it is preferable to make a copy of `out.len()` before moving instead I'll change that.